### PR TITLE
DEV: uses swipe-events lib for swipe modifier

### PR DESCRIPTION
**Pull request by jjaffeux**
This commit also:

- uses the swipe modifier in the glimmer-site-header component
- changes closing condition for d-modal and toast from distance to velocity
- cancels toast auto close on touch

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/d-modal.gjs [hunk 1]
```diff
@@ -121,39 +121,36 @@ export default class DModal extends Component {
   }
 
   @action
-  async handleSwipe(state) {
-    if (!this.site.mobileView) {
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/components/d-modal.gjs:125]: this is handled at the modifier level with à@enabled={{this.site.mobileView}}`

File: app/assets/javascripts/discourse/app/lib/swipe-events.js [hunk 2]
```diff
@@ -1,6 +1,52 @@
...
+  } else if (origin === "top") {
+    return (
+      (e.deltaY < -SWIPE_DISTANCE_THRESHOLD &&
+        e.velocityY < SWIPE_VELOCITY_THRESHOLD) ||
+      e.velocityY < 0
+    );
+  }
+  return false;
+}
+
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/lib/swipe-events.js:49]: Not sure swipe-events is the right home for this, but it's already better than in the events logic where it was before

File: app/assets/javascripts/float-kit/addon/components/d-default-toast.gjs [hunk 3]
```diff
@@ -1,70 +1,63 @@
...
-                  @action={{fn
-                    toastAction.action
-                    (hash data=@data close=@close)
-                  }}
-                  class={{toastAction.class}}
-                  tabindex="0"
-                />
-              {{/if}}
-            {{/each}}
+const DDefaultToast = <template>
```
Comment (jjaffeux) [inline app/assets/javascripts/float-kit/addon/components/d-default-toast.gjs:8]: That's not really related, I just changed this to a template only component

File: app/assets/javascripts/float-kit/addon/modifiers/auto-close-toast.js [hunk 4]
```diff
@@ -33,6 +33,9 @@ export default class AutoCloseToast extends Modifier {
     this.duration = duration;
     this.timeRemaining = duration;
     this.progressBar = progressBar;
+    this.element.addEventListener("touchstart", this.stopTimer, {
```
Comment (jjaffeux) [inline app/assets/javascripts/float-kit/addon/modifiers/auto-close-toast.js:36]: We don't want the toast to disappear while touching it

File: app/assets/javascripts/discourse/app/components/d-modal.gjs (hunk 1 above)
Comment (CvX) [reply app/assets/javascripts/discourse/app/components/d-modal.gjs:125]: Would it make sense to move the mobileView check into `swipe` modifier itself? (i.e. are there cases where we want swipe enabled on desktop?)

**Review by CvX — APPROVED** [review]

**Comment by jjaffeux** [comment]
That's the case :D but I forgot this case

**Comment by jjaffeux** [comment]
> Would it make sense to move the mobileView check into swipe modifier itself? (i.e. are there cases where we want swipe enabled on desktop?)

You right it will always be enabled on mobile

**Pull request merged**